src/compiler/utilities.ts

```diff
@@ -3837,8 +3837,8 @@ namespace ts {
     // There is no reason for this other than that JSON.stringify does not handle it either.
     const doubleQuoteEscapedCharsRegExp = /[\\\"\u0000-\u001f\t\v\f\b\r\n\u2028\u2029\u0085]/g;
     const singleQuoteEscapedCharsRegExp = /[\\\'\u0000-\u001f\t\v\f\b\r\n\u2028\u2029\u0085]/g;
-    // Template strings should be preserved as much as possible
-    const backtickQuoteEscapedCharsRegExp = /[\\`]/g;
+    // Template strings preserve simple LF newlines, still encode CRLF (or CR)
+    const backtickQuoteEscapedCharsRegExp = /\r\n|[\\\`\u0000-\u001f\t\v\f\b\r\u2028\u2029\u0085]/g;

```

Comment:
Can you add a test for this change? I'm unclear on why we wouldn't want to preserve these. `backtickQuoteEsacpedCharsRegExp` is only used when the quote character is `CharacterCodes.backtick`, which mean's we're preserving the raw text of the template string as much as possible, so it seems like adding extra escapes would break anyone using the raw strings.

Reply:
* I'll see if I can add a test for this.

* Re the reason, I listed two reasons in the commit log:
  - Not escaping things like a TAB or a FF character can get mangled by editor style settings
  - Not escaping a CRLF newline and instead having the template literal have a CRLF will be interpreted as a regular newline, and result in a string with just LF.  (It's even more likely to get mangled by your editor, but even if not you'll get that problem.)

Reply:
> - Not escaping things like a TAB or a FF character can get mangled by editor style settings

Yeah, I can see that being a possible issue, however I'm still not sure `"<TAB>foo"` should become `` `\tfoo` `` since it is still semantically different between the raw and cooked forms. There are only two ways to get to this code path: 1) Via a user-defined transformation, in which case we would want to preserve what was written, or 2) via one of our code fixes. If we want to normalize TAB or FF in the code fix, that's fine, but I'm concerned whether this would affect anyone using custom transforms. This is a small enough corner case its probably reasonable however, and we can address it if there's feedback.

> - Not escaping a CRLF newline and instead having the template literal have a CRLF will be interpreted as a regular newline, and result in a string with just LF. (It's even more likely to get mangled by your editor, but even if not you'll get that problem.)

_StringLiteral_ can't contain a raw CRLF unless its part of a _LineContinuation_, which becomes an empty code sequence, per NOTE 2 of https://tc39.es/ecma262/#sec-literals-string-literals and in https://tc39.es/ecma262/#sec-static-semantics-sv, so that wouldn't matter in a _StringLiteral_ → _TemplateLiteral_ conversion. The escape sequence `\r\n` in a string literal should remain as the escape sequence `\r\n` in the template literal, but we shouldn't just convert a raw CRLF into the escape sequence as that could impact custom transforms (which I *could* see taking a reliance on the CRLF->LF normalization of template literals). 

So, -1 on converting CRLF into the escaped form, +1 on converting other characters (at least until someone raises an issue).